maven-scm-providers/maven-scm-providers-git/maven-scm-provider-gitexe/src/main/java/org/apache/maven/scm/provider/git/gitexe/command/GitCommandLineUtils.java

```diff
@@ -196,4 +209,27 @@ public static int execute( Commandline cl, CommandLineUtils.StringStreamConsumer
         return exitCode;
     }
 
+    static Map<String, String> prepareEnvVariablesForRepository( GitScmProviderRepository repository,
+                                                                 Map<String, String> environmentVariables )
+    {
+        Map<String, String> effectiveEnvironmentVariables = new HashMap<>();
+        if ( environmentVariables != null )
+        {
+            effectiveEnvironmentVariables.putAll( environmentVariables );
+        }
+        if ( StringUtils.isNotBlank( repository.getPrivateKey() ) )
+        {
+            if ( effectiveEnvironmentVariables.putIfAbsent( VARIABLE_GIT_SSH_COMMAND, "ssh -IdentitiesOnly=yes -i "
```

Comment:
`-o Identities...`?

Reply:
Good catch, fixed in https://github.com/apache/maven-scm/pull/159/commits/42722d952a03b2dc80e981d12c8c5aa8dba3cf87.